app/src/main/java/com/aclass/edx/helloworld/AudioRecorderActivity.java

```diff
@@ -176,8 +173,9 @@ private void startRecordingIfPermissionGranted() throws IOException {
     }
 
     private void startRecording() throws IOException {
-        audioFilename = Environment.getExternalStorageDirectory().getAbsolutePath() + "/";
-        audioFilename += "t4-" + generateRandomFilename() + ".3gp";
+        audioFilename = String.format("%s/%s.3gp",
+                getExternalCacheDir().getAbsolutePath(),
+                "t4-" + ActivityUtils.generateRandomString());
```

Comment:
No. This is just a placeholder method. I was planning to delete this activity from the MVP. This was just a test class for the InterviewActivity (which does not use a randomly generated filename). Do you think we should keep this?